src/icmp/pinger.cc

```diff
@@ -233,14 +231,13 @@ main(int, char **)
             icmp4.Recv();
         }
 
-        if (PINGER_TIMEOUT + last_check_time < squid_curtime) {
+        const auto waited = std::chrono::duration_cast<std::chrono::seconds>(timer.total()).count();
+        if (waited >= PINGER_TIMEOUT) {
```

Comment:
To keep code safer during future refactoring and copying, please do not downgrade to unitless types unless absolutely necessary. It is not necessary here:

```suggestion
        const auto seconds = std::chrono::duration_cast<std::chrono::seconds>(timer.total());
        if (seconds.count() >= PINGER_TIMEOUT) {
```

And then use `seconds.count()` in the debugs() as well.

I do not insist on the `seconds` variable name. `secondsWaited` also works well. `waitTime` and `delay` would work but they are not as readable with `.count()` suffixes. I would avoid `waited` because that sounds like a boolean variable.

FWIW, these `.count()` suffixes are only a sign of an unfinished conversion. They would go away if we fix `PINGER_TIMEOUT` type and add a printer for time durations (as discussed in earlier change requests). I do not insist on those improvements -- just pointing out that the remaining problems disappear with complete/proper conversion. Until that happens, please resist the temptation to downgrade to unitless types.

Reply:
Ah, sorry I mixed up time_point issues with duration. Testing the suggested change now.

Reply:
This specific change done in [a70f527](https://github.com/squid-cache/squid/pull/1277/commits/a70f52722a00314490c478cf3ef1fad9a70e00f8)